aiohttp/connector.py

```diff
@@ -964,6 +980,15 @@ async def _wrap_create_connection(
             async with ceil_timeout(
                 timeout.sock_connect, ceil_threshold=timeout.ceil_threshold
             ):
+                if self._network_interface:
+                    args, sock_hostname, sock_port = args
```

Comment:
This isn't right, it will error if `*args` is not of length 3.
If we can rely on it always having 3 items, then we should just turn them into positional parameters and drop this line entirely.

At a brief glance this does appear to be a private function that is always passed `factory, host, port`, so it may be a good option to change the parameter.

Reply:
Is this acceptable? This will allow more parameters in the future if they are added in the front.
```python
*_, sock_hostname, sock_port = args
args = args[:-2]
```

Reply:
Use an actual name for the data. `args` is too generic. Also, it's better to have an atomic instruction rather than multiple, manipulating unidentified bits of information.

Reply:
I decided to go back to your original code suggestion and adding the specific type ignore rule. I have geninuely no idea how to make it more clean and atomic (without introducing unnecessary if-statements and extra vars)  than what I suggested til this point.